### DISCUSSION TITLE
Have Option To Redirect To A Different Destination Instead Of Default 404 Page

### DISCUSSION THREAD

**Discussion by dev-techguy** [ash-jc-allen/short-url#201] [Ideas]
Hello, have a way when a link is deactivated one can also be able to add a redirect link instead of the 404 page.

**Comment by ash-jc-allen** [comment]
Hey! Sorry for the slow reply. Currently, the package doesn't have any way of handling this out-of-the-box. Off the top of my head, I think I'd try adding a custom middleware class and registering it in your `config/short-url.php` file. In this middleware class, you could maybe try and check if the `ShortURL` is deactivated and then handle the redirect from there.

However, I think with this approach, you wouldn't be able to track the visits to the deactivated URLs because they wouldn't make it to the controller. So I'm not sure if this'll be any help for you?

Another approach you could take would be to override the `ShortURLController` and add your own custom route (https://github.com/ash-jc-allen/short-url#custom-route) pointing to your own controller. You can add your custom handling in that controller. The only issue I see with this approach is that you'll need to keep an eye out for any changes to the package's controller in future releases (to keep them in line with each other as close as possible). But, in all honesty, I don't imagine I'll be making any changes to it any time soon.

Hopefully one of these suggestions will help out a little bit? Feel free to ask any more questions you might have though 🙂

**Reply by dev-techguy** [reply]
Thanks that worked
